### PR TITLE
HBASE-27235 Clean up error-prone warnings in hbase-hadoop-compat

### PR DESCRIPTION
**Pull request by apurtell**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 42s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 29s |  master passed  |
| +1 :green_heart: |  compile  |   0m 11s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m  0s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 11s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  8s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 12s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 12s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m  0s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 30s |  hbase-hadoop-compat in the patch passed.  |
|  |   |  15m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4648/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4648 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 9673792f0c1f 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b1706a8a12 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4648/1/testReport/ |
| Max. process+thread count | 254 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat U: hbase-hadoop-compat |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4648/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m  1s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  1s |  master passed  |
| +1 :green_heart: |  compile  |   0m 14s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 48s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 33s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 14s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 14s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 46s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 37s |  hbase-hadoop-compat in the patch passed.  |
|  |   |  18m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4648/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4648 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 5a00346d8f45 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b1706a8a12 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4648/1/testReport/ |
| Max. process+thread count | 274 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat U: hbase-hadoop-compat |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4648/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 58s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 27s |  master passed  |
| +1 :green_heart: |  compile  |   0m 23s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 14s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 56s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 27s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 39s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 19s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 19s |  hbase-hadoop-compat generated 0 new + 2 unchanged - 51 fixed = 2 total (was 53)  |
| +1 :green_heart: |  checkstyle  |   0m  7s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  14m  6s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotless  |   0m 44s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 27s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   |  31m 16s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4648/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4648 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 3e2592dbccc1 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b1706a8a12 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 64 (vs. ulimit of 30000) |
| modules | C: hbase-hadoop-compat U: hbase-hadoop-compat |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4648/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-hadoop-compat/src/main/java/org/apache/hadoop/hbase/CompatibilitySingletonFactory.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
  */
 @InterfaceAudience.Private
 public class CompatibilitySingletonFactory extends CompatibilityFactory {
+
+  @SuppressWarnings("ImmutableEnumChecker")
```
Comment (Apache9) [inline hbase-hadoop-compat/src/main/java/org/apache/hadoop/hbase/CompatibilitySingletonFactory.java:35]: Not a good design but anyway, it works. Not your fault, let's file another issue to rewrite this...

**Review by Apache9 — APPROVED** [review]

**Pull request merged**